hetmatpy/pipeline.py

```diff
@@ -8,6 +8,22 @@
 import hetmatpy.degree_weight
 import hetmatpy.hetmat
 
+FLOAT_ERROR_TOLERANCE = 1e-5
+
+
+def calculate_sd(sum_of_squares, unsquared_sum, number_nonzero):
+    """
+    Calculate the standard deviation and validate the incoming data
+    """
+    if number_nonzero == 1:
+        return 0
+    # If all the values in the row are the same we'll manually return zero,
+    # because not doing so can lead to some issues with float imprecision
+    elif abs(sum_of_squares - unsquared_sum ** 2 / number_nonzero) < FLOAT_ERROR_TOLERANCE:
```

Comment:
Should you assign this value to `numerator` or something to prevent the duplicate computation below? These functions get called a LOT, so efficiency is a consideration